python/samples/getting_started/agents/azure_ai/azure_ai_with_azure_ai_search.py

```diff
@@ -18,6 +20,38 @@
 """
 
 
+def extract_citations_from_response(response: AgentRunResponse) -> list[dict[str, Any]]:
+    """Extract citation information from an AgentRunResponse."""
+    citations: list[dict[str, Any]] = []
+
+    if hasattr(response, "messages") and response.messages:
```

Comment:
you can skip a bunch of these checks (it's a sample, so we can be a less strict on typing, to improve readability) and inverse the logic for the rest:

```suggestion
    from itertools import chain
    
    if not response.messages:
        return citations
    for content in chain.from_iterable(message.contents for message in response.messages):
        if not content.annotations:
            continue
        # parse the annotations
```
Also why not just return the citations instead of this dict?